assets/source/setup-guide/app/components/Account/BusinessAccountSelection.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import { Spinner } from '@woocommerce/components';
+import { addQueryArgs } from '@wordpress/url';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	CardBody,
+	Flex,
+	FlexBlock,
+	FlexItem,
+	SelectControl,
+	__experimentalText as Text, // eslint-disable-line @wordpress/no-unsafe-wp-apis --- _experimentalText unlikely to change/disappear and also used by WC Core
+} from '@wordpress/components';
+
+const BusinessAccountSelection = ( {
+	businessAccounts,
+	setAttemptedCreation,
+} ) => {
+	const [ targetBusinessId, setTargetBusinessId ] = useState(
+		undefined !== businessAccounts && businessAccounts.length > 0
```

Comment:
If we can ensure that `businessAccounts` is always an array, then we don't need to check for undefined?

Reply:
`undefined` state is useful here to signify loading. An empty array is a different state.